game/activity.rpy

```diff
@@ -56,53 +61,207 @@ init python in jn_activity:
         twitter = 10
         deviantart = 11
         manga = 12
+        ddlc_moe = 13
```

Comment:
I agree - the only question I have would be around the manager class. Would it be more pythonic to use a staticmethod, and keep things under one class, or just as you described?

Reply:
You could keep the map I described in the `JNActivity` class, construct the activities directly, as well as implement the `check` method for it. Technically it would still be correct.

But *personally* I think it's ""more correct"" to keep the logic separate in this case. Looks like a perfect place to use the manager pattern. As activities don't *manage* themselves, it's the activity manager that *manages* them. It's a bit more flexible, for example you can have multiple managers with different reactions, or you can unregister activities from the managers, or override existing activities, or disallow overriding them. It also means we can change the logic in one class, but  keep in the other: e.g. if we added a subclass of `JNActivity` which selects the message based on your affinity. We wouldn't need any changes in the manager, or they would be minimal.

I leave the final decision to you, though.

Reply:
No worries - I'll crack on with it